daffodil-core/src/main/scala/org/apache/daffodil/dsom/ModelGroup.scala

```diff
@@ -130,6 +130,22 @@ object TermFactory {
   }
 }
 
+trait SharedGroupMembersFactoryMixin { sset: SchemaSet =>
+
+  private type SharedType = Seq[Term]
+
+  private val f = new SharedFactory[SharedType]
+
+  /**
+   * Evaluates the argument only if passed a new key it has never seen before.
+   *
+   * If it has seen the key, it returns the
+   * initial sharedArg evaluation from the first time it encountered them.
+   */
+  final def sharedGroupMembersFactory(key: SharedFactory.ShareKey, sharedArg: => SharedType) =
```

Comment:
We're using a private type alias in the signature of a public function?

This wont cause any problems (as Scala considers them to be equivalent), but seems confusing.

In this case, it is not clear to me that aliasing the type to SharedType actually clarifies much to the reader; as both places it is used already have "shared" in their name.

Reply:
Yeah. I simplified this and got rid of these named members, or rather I jsut define them on SchemaSet directly rather than this mixin stuff.